lib/libspl/include/sys/isa_defs.h

```diff
@@ -100,28 +100,39 @@ extern "C" {
 
 #define	_SUNOS_VTOC_16
 
-/* arm arch specific defines */
-#elif defined(__arm) || defined(__arm__) || defined(__aarch64__)
+/* aarch64 arch specific defines */
+#elif defined(__aarch64__)
 
 #if !defined(__arm)
 #define	__arm
 #endif
 
-#if !defined(__arm__)
-#define	__arm__
-#endif
-
 #if defined(__aarch64__)
 #if !defined(_LP64)
 #define	_LP64
 #endif
+#endif
+
+#if defined(__AARCH64EL__)
+#define	_LITTLE_ENDIAN
 #else
+#define	_BIG_ENDIAN
+#endif
+
+#define	_SUNOS_VTOC_16
+
+/* aarch32 arch specific defines */
+#elif defined(__arm) || defined(__arm__)
+
```

Comment:
probably need to define __arm__ also.

 #if !defined(__arm__)
#define	__arm__
#endif

Reply:
right!